pkg/bundle/internal/target/target.go

```diff
@@ -128,20 +128,22 @@ func (r *Reconciler) syncConfigMap(
 	}
 	binData := resolvedBundle.BinaryData
 
+	annotations, labels := mapsFromBundle(bundleTarget.ConfigMap.Metadata)
+	annotations[trustapi.BundleHashAnnotationKey] = bundleHash
+
 	// If the resource exists, check if it is up-to-date.
 	if !apierrors.IsNotFound(err) {
 		// Exit early if no update is needed
-		if exit, err := r.needsUpdate(ctx, target.Kind, targetObj, bundle, bundleHash); err != nil {
+		if exit, err := r.needsUpdate(ctx, target.Kind, targetObj, bundle, bundleHash, annotations, labels); err != nil {
```

Comment:
```suggestion
		if exit, err := r.needsUpdate(ctx, target.Kind, targetObj, bundle, bundleHash, bundleTarget.ConfigMap.Metadata); err != nil {
```

Reply:
As you mentioned later on, ended up not needing this as WithAnnotations(annotations) handles this in conjunction with the added logic to the hash